<launch>
    <node name="serial_node"        pkg="rosserial_python"      type="serial_node.py">
        <param name="port"              type="string"               value="/dev/ttyACM0"/>
        <param name="baud"              type="int"                  value="57600"/>
    </node>  
    <!-- เรียกใช้การส่งข้อมูลจาก Arduino มายัง ROS  -->
    <node name="odometry_publisher" pkg="learning_tf" type="odom2.py" output="screen"></node>
    <!-- เปิดใช้งาน Wheel Odometry สำหรับเก็บข้อมูลการเคลื่อนที่เพื่อบันทึกค่าการเคลื่อนที่ -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find learning_tf)/config2.rviz"/>
    <!-- เปิดใช้งาน RVIZ simulation เพื่อแสดงการเคลื่อนที่จากข้อมูลการคเลื่อนที่แบบ Realtime -->
    <node name="rqt_gui" pkg="rqt_gui" type="rqt_gui"/>
    
</launch>
